clean up project structure

ryanlerch opened the pull request.

ryanlerch committed:
convert to Python 3

Run the 2to3 converter,and update the requirements.

then apply the following fixes from Nils Philippsen:

 * don't mechanically wrap iterables into lists
 * Checking if a list is empty and slicing it is easy:
   The `meetings` variable already holds a list (because it's the result of
   sorted(...)). I imagine at some point this was a generator or something,
   but with this, it's much easier to test if there are meetings and to
   only display the first couple of them.

Co-authored-by: Nils Philippsen <nils@redhat.com>
Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
Add some fallback if FAS credentials arent provided

This will move to noggin soon, but just add this for
now so the test passes

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
Fix flake8 issues

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
Try to fix the CI with the .cico.pipeline from noggin

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

abompard commented on pyproject.toml:
```diff
@@ -0,0 +1,61 @@
+"Fedora" = "supybot_fedora"
+
+[tool.black]
+skip-string-normalization = true
```
I think you can leave that out for new projects.

ryanlerch replied:
removed the whole black stanza from here

abompard commented on supybot_fedora/plugin.py:
```diff
@@ -311,8 +335,7 @@ def yield_github_repos(self, username):
         self.log.info("Finding github pull requests for %r %r" % (username, repo))
-        tmpl = "https://api.github.com/repos/{username}/{repo}/" + \
-            "pulls?per_page=100"
+        tmpl = "https://api.github.com/repos/{username}/{repo}/" + "pulls?per_page=100"
```
You could merge those two strings, that's probably a black artefact.

ryanlerch replied:
thanks! merged these together

abompard commented on supybot_fedora/plugin.py:
```diff
@@ -721,11 +767,12 @@ def swedish(self, irc, msg, args):
 
         # Import this here to avoid a circular import problem.
-        from __init__ import __version__
+        from .__init__ import __version__
```
It would be `from . import __version__`, it's a bit unusual to import from `__init__`

ryanlerch replied:
fixed.

abompard commented on supybot_fedora/plugin.py:
```diff
@@ -998,8 +1052,7 @@ def get_persons():
 
-        url = "https://apps.fedoraproject.org/" + \
-            "calendar/release-engineering/"
+        url = "https://apps.fedoraproject.org/" + "calendar/release-engineering/"
```
You could merge those strings

ryanlerch replied:
fixed

abompard commented on supybot_fedora/plugin.py:
```diff
@@ -1202,7 +1263,7 @@ def quote(self, irc, msg, args, arguments):
         del symbols['ANN']  # And this one is unused...
 
-        key_fmt = lambda d: ', '.join(sorted(d.keys()))
+        key_fmt = lambda d: ', '.join(sorted(d.keys()))  # noqa: E731
```
We're defining quite a bit of lambda functions like this, we could move all that to a separate module (and `cmp` too), or properly define them with a `def` at the top of the file, no?

ryanlerch replied:
yeah, i wanted to kind of leave this level of restructuring out of this PR, but my intent was to fix these afterwards.

filed the following ticket so i don't forget:
https://github.com/fedora-infra/supybot-fedora/issues/67

ryanlerch committed:
Run Black over the code

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
Use Poetry

rearrange everything, and use poetry.

note that the tests still fail, because we are trying to contact FAS
without a username and password configured.

Note too, this removes (or comments out) fedmsg code, since we are
going to port this over very soon.

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

abompard commented on pyproject.toml:
```diff
@@ -0,0 +1,58 @@
+[tool.poetry]
+name = "supybot-fedora"
+version = "0.0.1"
```
Apparently the packaging & release process was done in the Makefile. The version is set there to be `0.3.5`, I think that's what we should use.

ryanlerch replied:
done!

abompard commented on requirements.txt:
```diff
@@ -1,9 +1,10 @@
 python-fedora>=0.3.7
```
Now that the dependencies are in `pyproject.toml`, you can remove this file.

ryanlerch replied:
done!

abompard commented:
Also, you'll probably have to redo the black run, since it will now convert the string markers.

ryanlerch commented:
> Also, you'll probably have to redo the black run, since it will now convert the string markers.

Done!

abompard reviewed (approved):
LGTM, but how are we supposed to test this?
